src/modules/user/services/wallet.service.ts

```diff
@@ -0,0 +1,85 @@
+import { Injectable } from "@nestjs/common";
+import { InjectRepository } from "@nestjs/typeorm";
+import { Repository } from "typeorm";
+import { utils } from "ethers";
+
+import { Wallet } from "../model/wallet.entity";
+import { User } from "../model/user.entity";
+import { InvalidSignatureException, UserNotFoundException, WalletNotFoundException } from "./exceptions";
+
+@Injectable()
+export class WalletService {
+  constructor(
+    @InjectRepository(Wallet) private walletRepository: Repository<Wallet>,
+    @InjectRepository(User) private userRepository: Repository<User>,
+  ) {}
+
+  public async upsertWallet({ userId, walletAddress }: { userId: number; walletAddress: string }) {
+    await this.assertUserExists(userId);
+
+    let wallet = await this.walletRepository.findOne({
+      where: {
+        userId,
+      },
+    });
+
+    if (!wallet) {
+      wallet = this.walletRepository.create({
+        userId,
+        walletAddress,
+      });
+    } else {
+      wallet.walletAddress = walletAddress;
+    }
+
+    wallet = await this.walletRepository.save(wallet);
+
+    return wallet;
+  }
+
+  public verifySignedDiscordId({
+    signature,
+    discordIdMessage,
+    walletAddress,
+  }: {
+    signature: string;
+    discordIdMessage: string;
+    walletAddress: string;
+  }) {
+    const recoveredAddress = utils.verifyMessage(discordIdMessage, signature);
+
+    if (recoveredAddress.toLowerCase() !== walletAddress.toLowerCase()) {
+      throw new InvalidSignatureException();
+    }
+  }
+
+  public async deleteWalletByUserId(userId: number) {
+    return this.walletRepository.delete({
+      userId,
+    });
+  }
+
+  public async assertWalletForUserExists(userId: number) {
+    const wallet = await this.walletRepository.findOne({
+      where: {
+        userId,
+      },
+    });
+
+    if (!wallet) {
+      throw new WalletNotFoundException(userId);
+    }
+  }
+
+  public async assertUserExists(userId: number) {
+    const user = await this.userRepository.findOne({
+      where: {
+        id: userId,
+      },
+    });
+
+    if (!user) {
+      throw new UserNotFoundException();
+    }
+  }
```

Comment:
I think we remove this method and use the existing:
https://github.com/across-protocol/scraper-api/blob/847f4304a754a16355d44cc075f5f9257dd23a76/src/modules/user/services/user.service.ts#L26

Reply:
done